src/ai/backend/manager/services/resource_preset/service.py

```diff
@@ -101,7 +106,11 @@ async def modify_preset(
         modifier = action.modifier
 
         if preset_id is None and name is None:
-            raise ValueError("One of (`id` or `name`) parameter should be not null")
+            raise InvalidAPIParameters("One of (`id` or `name`) parameter should be not null")
+
+        if resource_slots := modifier.resource_slots.optional_value():
+            if not all(slot in resource_slots for slot in _INTRINSIC_RESOURCE_SLOTS):
+                raise InvalidAPIParameters("ResourceSlot must have all intrinsic resource slots.")
```

Comment:
Rather than checking this, it would be better to pass the value when the resource_slot is first created.

Reply:
Would it be better to throw an exception in `to_creator`, and `to_modifier`? 🤔 
I think throwing an exception in these funcs might make it more bothersome to test than the current code.

Reply:
My intention was to put the validation logic in the dataclass definition.